extensions/azurePublish-new/src/components/provisioningWizards/steps/ResourceConfigurationStep.tsx

```diff
@@ -0,0 +1,246 @@
+  IStackTokens,
+  IStackItemStyles,
+  Link,
+} from 'office-ui-fabric-react';
+
+import {
+  userInfoState,
+  resourceGroupSelectionState,
+  subscriptionSelectionState,
+  tenantSelectionState,
+} from '../../../recoilModel/atoms/resourceConfigurationState';
+import { ResourceGroupPicker } from '../../resourceConfiguration/ResourceGroupPicker';
+import { SubscriptionPicker } from '../../resourceConfiguration/SubscriptionPicker';
+import { TenantPicker } from '../../resourceConfiguration/TenantPicker';
+
+type Props = {
+  onResourceConfigurationChange: (isValidConfiguration: boolean) => void;
+};
+type ProvisonActionsStylingProps = {
+  showSignout: boolean;
+};
+const ProvisonActions = styled.div<ProvisonActionsStylingProps>((props) => ({
+  display: 'flex',
+  flexFlow: 'row nowrap',
+  justifyContent: props.showSignout ? 'space-between' : 'flex-end',
+}));
+
+const ConfigureResourcesSectionName = styled(Text)`
+  font-size: ${FluentTheme.fonts.mediumPlus.fontSize};
+  font-weight: ${FontWeights.semibold};
+  margin-bottom: 4px;
+`;
+
+const ConfigureResourcesSectionDescription = styled(Text)`
+  font-size: ${FluentTheme.fonts.medium.fontSize};
+  line-height: ${FontSizes.size14};
+  margin-bottom: 20px;
+`;
+
+const configureResourcePropertyStackTokens: IStackTokens = { childrenGap: 5 };
+
+const configureResourcePropertyLabelStackStyles: IStackItemStyles = {
+  root: {
+    width: '200px',
+  },
+};
+
+const ConfigureResourcesPropertyLabel = styled(Label)`
+  font-size: ${FluentTheme.fonts.medium.fontSize};
+  font-weight: ${FontWeights.regular};
+`;
+
+const configureResourceDropdownStyles = { root: { paddingBottom: '4px', width: '300px' } };
+
+const configureResourceTextFieldStyles = { root: { paddingBottom: '4px', width: '300px' } };
+
+const configureResourcesIconStyle = {
+  root: {
+    color: NeutralColors.gray160,
+    userSelect: 'none',
+  },
+};
+
+const LearnMoreLink = styled(Link)`
+  user-select: none;
+  font-size: 14px;
+`;
+export const ResourceConfigurationStep = (props: Props) => {
+  const [userInfo, setUserInfo] = useRecoilState(userInfoState);
+  const [tenantId, setTenantId] = useRecoilState(tenantSelectionState);
+  const [subscriptionId, setSubscriptionId] = useRecoilState(subscriptionSelectionState);
+  const [resourceGroupId, setResourceGroupId] = useRecoilState(resourceGroupSelectionState);
+
+  const { onResourceConfigurationChange } = props;
+
+  const isValidConfiguration = (): boolean => {
+    return true;
+  };
+
+  const handleTenantChange = (tenantId: string) => {
+    setTenantId(tenantId);
+    setSubscriptionId('');
+    setResourceGroupId('');
+  };
+
+  const handleSubscriptionChange = (subscriptionId: string) => {
+    setSubscriptionId(subscriptionId);
+    onResourceConfigurationChange(isValidConfiguration());
+  };
+
+  const handleResourceGroupChange = (resourceGroupId: string) => {
+    setResourceGroupId(resourceGroupId);
+    onResourceConfigurationChange(isValidConfiguration());
+  };
```

Comment:
nit: add new line after 120, there are a bunch of similar cases in the PR, keep it consistent